### PR TITLE
Reduce the prototype.js usage

### PR DESCRIPTION
**Pull request by Wadeck**
Only cover the Element.xxxClassName in hudson-behavior. The related methods can be easily replaced by vanilla JavaScript.
For compatibility: https://caniuse.com/classlist (add, remove and contains with single argument is supported by all supported browsers)

Manually tested the build history with more than 25 items to have the "pagination" working.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX). No ticket, too small adjustment + no need backport.

### Proposed changelog entries

* (internal) Replace the Element.xxxClassName methods by vanilla JavaScript.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

cc @jvz (for his interest in Vanilla)

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by res0nance — APPROVED** [review]
Looks reasonable

**Review by timja — APPROVED** [review]
great initiative, 💥

**Review by jvz — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (zbynek) [inline war/src/main/webapp/scripts/hudson-behavior.js:2223]: Is it certain that `rows[0]` is not undefined?

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (zbynek) [inline war/src/main/webapp/scripts/hudson-behavior.js:1812]: Is it certain that `el1` and `el2` are not undefined?

**Review by zbynek — COMMENTED** [review]
The Element methods were null pointer safe, the new code is not. In most cases the elements cannot be null because the code would fail elsewhere, but there are some places where we may want to check for truthy values explicitly.

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js:625]: Could this be replaced by something like `clsName.split(' ').forEach(function (className) { if (className) be.classList.add(className) })` ?

**Review by fqueiruga — APPROVED** [review]
LGreatTM

**Review by alecharp — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (varyvol) [reply war/src/main/webapp/scripts/hudson-behavior.js:2223]: It must be given it was also used before?

**Review by varyvol — APPROVED** [review]

**Comment by alecharp** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
